### ISSUE TITLE
Property flow state wins over [NotNull] attribute for warnings

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#37993]
Slightly related to #37313

**Version Used**: a08336f

**Steps to Reproduce**:

```cs
namespace ConsoleApp114
{
#nullable enable

    public class Doot
    {
        private string _toot = string.Empty;

        [AllowNull, NotNull]
        public string Toot
        {
            get => _toot;
            set => _toot = value ?? string.Empty;
        }
    }
    class Program
    {

        static void Main(string[] args)
        {
            var doot = new Doot();
            doot.Toot = null; // No warning, as expected
            Console.Write(doot.Toot.Length); // Unexpected warning
        }
    }
}
```

/cc @jcouv
